### ISSUE TITLE
Bug: toSVG() not working as expected in nodejs

### ISSUE THREAD

**Issue by willKo** [fabricjs/fabric.js#4355]
## Version
1.7.4

## Test Case
https://github.com/willKo/BugTest.git

## Steps to reproduce
 - npm install 
 - npm start 
 - dataurlPNG.txt and test.svg in browser 
## Expected Behavior
Characters in SVG are correctly postioned as e.g. in the png
## Actual Behavior
Characters in SVG are not postioned as e.g. in png. 

## Assumed Bug
 Custom Font was used to demermine Text position in toSVG() method. 

## Description 
I try to use fabricjs  on a node server to create svgs. 
My problem is that the toSVG() function of the canvas does not recognize the fonts correctly and calculates the character positions according to the wrong font. 

used Versions: 
```
 node: 6.10
 canvas: 1.6.5
 fabric: 1.7.14
```
 AFAIK: Newer '1.7.*' versions of  fabric, also do not calculate the positions correctly. 

It is working for PNGs according to the tutorial (http://fabricjs.com/fabric-intro-part-4)-

```
   canvas.contextContainer.addFont(font);
      canvas.contextTop.addFont(font);
      canvas.toDataURL({
        format: 'png',
      })
    );

```
And the png output is correctly positioned (see example), but using: "toSVG()" does not apply the (custom-) fonts correctly. 

In the example you can see that custom fonts are not displayed correctly.
The correct font is used for display, because the fonts are installed on my current system,  but not on the node server that creates the svg!) 

Is there anything like "contextTop"/"contextContainer" where I can add custom fonts, so they are  applied in the toSVG() function? 

** For example image see other Posts!**

## Other Posts 
-  https://stackoverflow.com/questions/46468609/fabricjs-nodejs-tosvg-wrongtext-positioning
-  https://groups.google.com/forum/#!topic/fabricjs/mff7Pr4fkGY

**Comment by willKo** [comment]
With some help and debugging I managed to include the fonts in the document:

https://github.com/willKo/BugTest/tree/includeFontsInDOM

Still the positioning of the Characters is wrong! 
It seems to vary by operating system, node version, installed GTK /(cairo-) version and/or fabricjs version  (not sure). 
Also wrong postioning in current fabric version on latest nodejs image on debian os. 

I normally test it with my image on dockerhub: 
https://hub.docker.com/r/will890/docker-node-serverless/ 
(in case someone needs a linux distribution to clone and test the project).

**Comment by asturur** [comment]
So the real problem is general nodecanvas usage with custom font.
They are fixing it in latest version of nodecanvas, other than that lib pango will be required.
So using nodecanvas 2.0 when is out of alpha will definetely help on this.

**Comment by willKo** [comment]
Thank you for your response. 
So the second approach (adding a style tag to jsdoms head-element) is correct for adding fonts? 

```javascript
const style = fabric.document.createElement('style');
style.textContent = "@font-face {font-family: 'Economica-Regular';...}";
fabric.document.head.appendChild(style);
```

Ok than I'll wait for the node canvas 2.0 release and hope that this will fix the wrong positioning. 

Regards

**Comment by asturur** [comment]
Actual problem between fabricjs and nodecanvas is that i'm using jsdom 9.11 that should bind nodecanvas behind canvas automatically but it does not seems to work for me and i m kinda stuck on it. i let it go and i ll put myself back on it as soon as i can.

**Comment by asturur** [comment]
i have no idea if that approach is correct. if the fonts LOOKS good on the output, despite positioning, then they have been added to canvas. If they were already added to canvas before fabricjs started measure them, this i do not know. There is a nodecanvas api to load fonts, did you check it?

**Comment by willKo** [comment]
Well the postioning LOOKS **different**, and the Font is correctly in png without adding it to the contextTop and the other context. 
Therefore my assumption is that the font was "correctly" recognized and used. 

Yes I tried the node canvas API also the newest alpha version which has  a new method (Canvas.registerFont). Unfortionaltely I think I did not configure the jsdom/nodecanvas correctly. 

My next approach is to wait for the release of version 2.0 of node-canvas and then I'll try again.
